include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -132,6 +132,22 @@ class MotorInterruptHandler {
         return false;
     }
 
+    auto homing_stopped() -> bool {
+        if (!can_step() && !limit_switch_triggered()) {
+            finish_current_move(AckMessageId::timeout);
+            return true;
+        }
+        if (can_step() && limit_switch_triggered()) {
```

Comment:
When you're homing, you don't know where you are. In practice, you'll be commanding a move that is longer than your motion extents. You _expect_, and want, to hit the limit switch _before_ you're done moving, because you don't know where you are. It's completely fine to hit the limit switch before you're done.